tvision/include/compatlayer.h

```diff
@@ -2108,7 +2109,9 @@ CLY_CFunc int  CLY_getcurdir(int drive, char *buffer);
  /* Platforms where sys/types.h doesn't define ssize_t: */
  #if defined(Uses_CLY_ssize_t) && !defined(CLY_ssize_t)
   #define CLY_ssize_t 1
-  typedef long ssize_t;
+	#ifndef _SSIZE_T_DEFINED
+		typedef long ssize_t;
+	#endif
```

Comment:
Salvador, I tried to follow your suggestion on the previous pull request to avoid defining Uses_CLY_ssize_t at line 560.  However, <sys/types.h> is only included immediately prior to here, around line 2108, so _SSIZE_T_DEFINED is not defined up at line 560.  I didn't want to start moving things around, since there are so many includes and dependencies between them.

Reply:
I just checked out the source for mingw_w64 from here: https://github.com/mingw-w64/mingw-w64

Going all the way back to version 1.x in 2009, mingw_w64 defined ssize_t. The older, original mingw has moved here: https://osdn.net/projects/mingw/.  It is still only a 32-bit compiler.  I don't think we should support it.  What is your opinion?  I have not checked this older original mingw to see if/when it first added ssize_t.

I will try moving the include of sys/types.h up above Uses_CLY_ssize_t at line 560.  If it doesn't cause too many problems on all the compilers we want to support, then I suppose that is the better solution.

Reply:
Why not just use a define like I suggest? 
About the old MinGW: which version was this? 
You should look for some definition that allows differentiating these versions. With this we can just avoid the Uses_CLY_ssize_t for modern MinGW versions.

Reply:
The original MinGW is here: https://sourceforge.net/projects/mingw/.  You'll see there that it has moved to a new location. MingW-w64 was forked from the original years ago, and has moved on, while the original project has not seen significant updates, though it continues to generate releases.  The original is still only 32-bit.  My opinion is we should test for the newer MingW-w64, and if it is not being used, abort the build and tell the user to use MingW-w64.  Do you have a particular reason for wanting to support the older 32-bit version?  I don't think many people are still using it.  Plus, we have a big enough challenge supporting just MingW-w64, along with other compilers (MSC, GCC).

Reply:
Please see latest commit, I believe I've addressed this issue.